app/src/lib/markdown-filters/node-filter.ts

```diff
@@ -104,15 +106,21 @@ export const buildCustomMarkDownNodeFilterPipe = memoizeOne(
  */
 export async function applyNodeFilters(
   nodeFilters: ReadonlyArray<INodeFilter>,
-  parsedMarkdown: string
-): Promise<string> {
-  const mdDoc = new DOMParser().parseFromString(parsedMarkdown, 'text/html')
+  markdownEmitter: MarkdownEmitter
+): Promise<void> {
+  if (markdownEmitter.latestMarkdown === null) {
+    return
+  }
+
+  const mdDoc = new DOMParser().parseFromString(
+    markdownEmitter.latestMarkdown,
+    'text/html'
+  )
 
   for (const nodeFilter of nodeFilters) {
     await applyNodeFilter(nodeFilter, mdDoc)
+    markdownEmitter.emit(mdDoc.documentElement.innerHTML)
   }
```

Comment:
Just a suggestion: since this can be kind of expensive depending on the markdown… maybe we could ad a way of breaking out of this loop if the emitter is disposed?

Reply:
Great suggestion. I figured also to put one at the top since this is hit after the markdown emitter is created during marked js parsing. That ensures that the markdown emitter at-least should be viable when we hit the first filter application.